RecoEgamma/EgammaHLTProducers/interface/EgammaHLTCombinedIsolationProducer.h

```diff
@@ -27,20 +27,20 @@ namespace edm {
   class ConfigurationDescriptions;
 }
 
-class EgammaHLTCombinedIsolationProducer : public edm::EDProducer {
+class EgammaHLTCombinedIsolationProducer : public edm::global::EDProducer<> {
 public:
   explicit EgammaHLTCombinedIsolationProducer(const edm::ParameterSet&);
   ~EgammaHLTCombinedIsolationProducer() override;
   
   static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
-  void produce(edm::Event&, const edm::EventSetup&) override;
+  void produce(edm::StreamID, edm::Event&, const edm::EventSetup&) const override;
 private:
   // ----------member data ---------------------------
   
   edm::EDGetTokenT<reco::RecoEcalCandidateCollection> recoEcalCandidateProducer_;
   std::vector<edm::EDGetTokenT<reco::RecoEcalCandidateIsolationMap> > IsolTag_;
   std::vector<double> IsolWeight_;
```

Comment:
Now I was also worried about vectors of non-const types, but fortunately this is not an issue. Calling a non-const member function of a vector element disregards the const qualifier of `produce`. Here an example which doesn't compile:

```
#include <vector>

class C {
  public:
    C() : c_(0) {}
    void increment() { ++c_; }
  private:
    int c_;
};

class A {
  public:
    A() {
        v_.push_back(C{});
    }
    void produce () const { v_[0].increment(); }
  private:
    std::vector<C> v_;
};

int main() {
    A a{};
    a.produce();
}
```

Reply:
@guitargeek ,  the point is that
```
   matcher_->setupES(iSetup);
```
can modify the object pointed by ```matcher_``` independently in all threads, something which is not allowed for a ```global``` module.